crates/oxc_linter/src/snapshots/no_empty_object_type.snap

```diff
@@ -0,0 +1,173 @@
+   ╭─[no_empty_object_type.tsx:6:4]
+ 5 │             
+ 6 │             interface Derived extends Base {}
+   ·             ─────────────────────────────────
+ 7 │             
+   ╰────
+  help: To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.
+
+  ⚠ typescript-eslint(no-empty-object-type): Do not use the empty object type literal.
+   ╭─[no_empty_object_type.tsx:6:4]
+ 5 │             
+ 6 │             interface Derived extends Base {}
+   ·             ─────────────────────────────────
+ 7 │                   
+   ╰────
+  help: To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.
+
+  ⚠ typescript-eslint(no-empty-object-type): Do not use the empty object type literal.
+   ╭─[no_empty_object_type.tsx:1:1]
+ 1 │ interface Base extends Array<number> {}
+   · ───────────────────────────────────────
+   ╰────
+  help: To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.
+
+  ⚠ typescript-eslint(no-empty-object-type): Do not use the empty object type literal.
+   ╭─[no_empty_object_type.tsx:1:1]
+ 1 │ interface Base extends Array<number | {}> {}
+   · ────────────────────────────────────────────
+   ╰────
+  help: To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.
+
+  ⚠ typescript-eslint(no-empty-object-type): Do not use the empty object type literal.
+   ╭─[no_empty_object_type.tsx:1:39]
+ 1 │ interface Base extends Array<number | {}> {}
+   ·                                       ──
+   ╰────
+  help: To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.
+
+  ⚠ typescript-eslint(no-empty-object-type): Do not use the empty object type literal.
+   ╭─[no_empty_object_type.tsx:5:4]
+ 4 │             }
+ 5 │             interface Base extends Array<Derived> {}
+   ·             ────────────────────────────────────────
+ 6 │                   
+   ╰────
+  help: To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.
+
+  ⚠ typescript-eslint(no-empty-object-type): Do not use the empty object type literal.
+   ╭─[no_empty_object_type.tsx:3:4]
+ 2 │             type R = Record<string, unknown>;
+ 3 │             interface Base extends R {}
+   ·             ───────────────────────────
+ 4 │                   
+   ╰────
+  help: To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.
+
+  ⚠ typescript-eslint(no-empty-object-type): Do not use the empty object type literal.
+   ╭─[no_empty_object_type.tsx:1:1]
+ 1 │ interface Base<T> extends Derived<T> {}
+   · ───────────────────────────────────────
+   ╰────
+  help: To avoid confusion around the {} type allowing any non-nullish value, this rule bans usage of the {} type.
```

Comment:
1. This span is a little too broad I think. the original rule highlights just the interface name, which seems like a good idea to me
2. we should have a separate error message for interfaces, since they are distinct from object literals

Reply:
could even just report on the `{}` (assuming its not too tricky to find